### PR TITLE
BOM-1043

### PR DESCRIPTION
**Pull request by awais786**
Implemented following.

- Isort
- Quality fixes
- Fixing django22 changes
- Added models `__str__` method and add few tests for verification
- Removed travis failures condition on python3
- Added test for admin.
- Replaced deprecated MIDDLEWARE_CLASSES setting with MIDDLEWARE
- twine

### CODE REVIEW

File: edxval/api.py [hunk 1]
```diff
@@ -300,6 +284,8 @@ def get_video_transcript_data(video_id, language_code):
             )
             raise
 
+    return None
```
Comment (awais786) [inline edxval/api.py]: fixing quality issue

File: edxval/utils.py [hunk 2]
```diff
@@ -180,10 +183,10 @@ def get_video_transcript_storage():
...
             settings.VIDEO_TRANSCRIPTS_SETTINGS.get('STORAGE_CLASS'),
         )(**settings.VIDEO_TRANSCRIPTS_SETTINGS.get('STORAGE_KWARGS', {}))
-    else:
-        # during edx-platform loading this method gets called but settings are not ready yet
-        # so in that case we will return default(FileSystemStorage) storage class instance
-        return get_storage_class()()
+
+    # during edx-platform loading this method gets called but settings are not ready yet
+    # so in that case we will return default(FileSystemStorage) storage class instance
+    return get_storage_class()()
```
Comment (awais786) [inline edxval/utils.py]: fixing quality

File: edxval/utils.py [hunk 3]
```diff
@@ -180,10 +183,10 @@ def get_video_transcript_storage():
         return get_storage_class(
             settings.VIDEO_TRANSCRIPTS_SETTINGS.get('STORAGE_CLASS'),
         )(**settings.VIDEO_TRANSCRIPTS_SETTINGS.get('STORAGE_KWARGS', {}))
-    else:
-        # during edx-platform loading this method gets called but settings are not ready yet
-        # so in that case we will return default(FileSystemStorage) storage class instance
-        return get_storage_class()()
+
```
Comment (awais786) [inline edxval/utils.py]: quality fix

File: edxval/tests/test_api.py [hunk 4]
```diff
@@ -852,7 +862,9 @@ def test_get_videos_for_ids_duplicates(self):
         self.assertEqual(len(videos), 1)
 
     def test_get_videos_for_ids_sort(self):
+        """ Tests for videos """
```
Comment (jmbowman) [inline edxval/tests/test_api.py]: If you regenerate `pylintrc` with a relatively recent version of `edx-lint`, it shouldn't require you to provide docstrings for test methods.

File: edxval/tests/test_api.py [hunk 5]
```diff
@@ -1063,6 +1075,7 @@ def get_child_tags(elem):
         for left_child, right_child in zip(left, right):
             self.assert_xml_equal(left_child, right_child)
 
+    # pylint: disable=c-extension-no-member
```
Comment (jmbowman) [inline edxval/tests/test_api.py]: Curious again what triggered this.

File: edxval/tests/test_api.py [hunk 6]
```diff
@@ -1170,10 +1183,13 @@ def test_transcript_export(self):
...
         # Verify transcript file is created.
-        self.assertEqual(sorted(transcript_files.values()), sorted(self.file_system.listdir(constants.EXPORT_IMPORT_STATIC_DIR)))
+        self.assertEqual(
+            sorted(transcript_files.values()),
+            sorted(self.file_system.listdir(constants.EXPORT_IMPORT_STATIC_DIR))
+        )
 
         # Also verify the content of created transcript file.
-        for language_code in transcript_files.keys():
+        for language_code in transcript_files.keys():  # pylint: disable=consider-iterating-dictionary
```
Comment (jmbowman) [inline edxval/tests/test_api.py]: That warning was actually good advice, this should probably just be `for language_code in transcript_files:`.

File: edxval/tests/test_api.py [hunk 7]
```diff
@@ -1196,7 +1211,7 @@ def test_unknown_video(self):
 
 
 @ddt
-class ImportTest(TestCase):
+class ImportTest(TestCase):  # pylint: disable=missing-docstring
```
Comment (jmbowman) [inline edxval/tests/test_api.py]: You may not need this if you updated `pylintrc` as suggested above.

File: edxval/transcript_utils.py [hunk 8]
```diff
@@ -1,16 +1,20 @@
 """
 A module containing transcripts utils.
 """
+# pylint: disable=inconsistent-return-statements
```
Comment (jmbowman) [inline edxval/transcript_utils.py]: I only see one place where a missing `return None` is probably triggering this, can we avoid the module-wide warning disable?

File: edxval/transcript_utils.py [hunk 9]
```diff
@@ -1,16 +1,20 @@
...
+
 from __future__ import absolute_import
+
 import json
-from six import text_type
 
-from pysrt import SubRipFile, SubRipItem, SubRipTime
+import six
+from pysrt import SubRipFile  # pylint: disable=wrong-import-order
+from pysrt import SubRipItem, SubRipTime
```
Comment (jmbowman) [inline edxval/transcript_utils.py]: Why were these imports from the same module split into two lines?

File: edxval/utils.py [hunk 10]
```diff
@@ -3,13 +3,16 @@
...
 import json
+
 from django.conf import settings
 from django.core.files.storage import get_storage_class
 from fs.path import combine
 from pysrt import SubRipFile
 
 
+# pylint: disable=missing-docstring
```
Comment (jmbowman) [inline edxval/utils.py]: Again, I'd rather avoid module-wide warning disables like this if we can.  The latest `pylintrc` from edx-lint should be a little less strict about which methods need docstrings.

File: edxval/utils.py [hunk 11]
```diff
@@ -208,10 +211,12 @@ def get_transcript_format(transcript_content):
         transcript_content (str): Transcript file content.
     """
     try:
-        sjson_obj = json.loads(transcript_content)
+        sjson_obj = json.loads(transcript_content)  # pylint: disable=unused-variable
```
Comment (jmbowman) [inline edxval/utils.py]: As the warning suggested, the `sjson_obj = ` part looks unnecessary.

File: edxval/utils.py [hunk 12]
```diff
@@ -208,10 +211,12 @@ def get_transcript_format(transcript_content):
...
     except ValueError:
         # With error handling (set to 'ERROR_RAISE'), we will be getting
         # the exception if something went wrong in parsing the transcript.
+
+        # pylint: disable=len-as-condition
```
Comment (jmbowman) [inline edxval/utils.py]: I think the warning was just suggesting that you could more simply replace `if len(srt_subs) > 0` with `if srt_subs`.

File: edxval/views.py [hunk 13]
```diff
@@ -125,6 +117,7 @@ def post(self, request):
         attrs = ('video_id', 'name', 'language_code', 'provider', 'file_format')
         missing = [attr for attr in attrs if attr not in request.data]
         if missing:
+            # pylint: disable=deprecated-method
             LOGGER.warn(
```
Comment (jmbowman) [inline edxval/views.py]: Another case where `LOGGER.warning` is preferred.

File: tox.ini [hunk 14]
```diff
@@ -18,7 +22,9 @@ deps =
 commands =
     pylint edxval
     pylint --py3k edxval
-    pycodestyle edxval
-    pydocstyle edxval
-    isort --check-only --recursive edxval manage.py setup.py
+    pycodestyle edxval --ignore=D100,D101,D102,D104,D105,D106,D107,D200,D202,D203,D204,D205,D210,D212,D213,D215,D400,D401,D403,D404,D405,D406,D407,D408,D409,D410,D411,D412,D413,D414
+    pydocstyle edxval --ignore=D100,D101,D102,D104,D105,D106,D107,D200,D202,D203,D204,D205,D210,D212,D213,D215,D400,D401,D403,D404,D405,D406,D407,D408,D409,D410,D411,D412,D413,D414
```
Comment (jmbowman) [inline tox.ini]: Could these ignores go in the appropriate config sections with some explanation, like in https://github.com/edx/edx-platform/blob/master/setup.cfg#L16-L33 ?

File: edxval/api.py [hunk 15]
```diff
@@ -207,6 +203,7 @@ def is_video_available(edx_video_id):
     return Video.objects.filter(edx_video_id=edx_video_id).exists()
 
 
+# pylint: disable=invalid-name
```
[the remark this answers is not in the document]
Comment (awais786) [reply edxval/api.py]:
```
************* Module edxval.api
C:206, 0: Function name "get_transcript_credentials_state_for_org" doesn't conform to u'([a-z_][a-z0-9_]{2,30}|test_[a-z0-9_]+)$' pattern (invalid-name)
C:235, 0: Function name "update_transcript_credentials_state_for_org" doesn't conform to u'([a-z_][a-z0-9_]{2,30}|test_[a-z0-9_]+)$' pattern (invalid-name)
C:302, 0: Function name "get_available_transcript_languages" doesn't conform to u'([a-z_][a-z0-9_]{2,30}|test_[a-z0-9_]+)$' pattern (invalid-name)
C:357, 0: Function name "create_or_update_video_transcript" doesn't conform to u'([a-z_][a-z0-9_]{2,30}|test_[a-z0-9_]+)$' pattern (invalid-name)
C:412, 0: Function name "get_3rd_party_transcription_plans" doesn't conform to u'([a-z_][a-z0-9_]{2,30}|test_[a-z0-9_]+)$' pattern (invalid-name)
C:435, 0: Function name "create_or_update_transcript_preferences" doesn't conform to u'([a-z_][a-z0-9_]{2,30}|test_[a-z0-9_]+)$' pattern (invalid-name)
C:665, 0: Function name "get_course_video_ids_with_youtube_profile" doesn't conform to u'([a-z_][a-z0-9_]{2,30}|test_[a-z0-9_]+)$' pattern (invalid-name)
C:765, 0: Function name "get_video_info_for_course_and_profiles" doesn't conform to u'([a-z_][a-z0-9_]{2,30}|test_[a-z0-9_]+)$' pattern (invalid-name)

[the remark this answers is not in the document]
Comment (jmbowman) [reply edxval/api.py]: The current `pylintrc` from `edx-lint` bumps the maximum length for function names to 40 characters, which should prevent all but two of these warnings.

**Pull request closed without merging**